target/linux/mediatek/filogic/base-files/etc/board.d/02_network

```diff
@@ -114,6 +115,10 @@ mediatek_setup_macs()
 		wan_mac=$(macaddr_add "$lan_mac" 1)
 		label_mac=$wan_mac
 		;;
+	wavlink,wl-wn586x3)
+		lan_mac=$(mtd_get_mac_text hw 0x460 17)
```

Comment:
Do you have plain text MAC in `hw` partition? You can try `mac-base` if you want. An example: https://github.com/openwrt/openwrt/blob/main/target/linux/ramips/dts/mt7621_tplink_ec330-g5u-v1.dts#L246